rest-api-spec/src/main/resources/rest-api-spec/test/cat.recovery/10_basic.yml

```diff
@@ -75,3 +75,59 @@
                 \n
               )+
               $/
+
+---
+"Test cat recovery output for closed index":
+  - skip:
+      version: " - 7.99.99"
+      reason: closed indices are replicated starting version 8.0.0
+
+  - do:
+      index:
+        index:  index2
+        id:     1
+        body:   { foo: bar }
+        refresh: true
+
+  - do:
+      indices.close:
+        index: index2
+  - is_true: acknowledged
+
+  - do:
+      cluster.health:
+        index: index2
+        wait_for_status: yellow
+
+  - do:
+      cat.recovery:
+        index: index2
+        h: i,s,t,ty,st,shost,thost,rep,snap,f,fr,fp,tf,b,br,bp,tb,to,tor,top
+
+  - match:
+      $body: |
+        /^
+        (
+          index2      \s+
+          \d          \s+                                 # shard
+          (?:\d+ms|\d+(?:\.\d+)?s)    \s+                 # time in ms or seconds
+          existing_store \s+                              # source type (always existing_store for closed indices)
+          done        \s+                                 # stage
+          [-\w./]+    \s+                                 # source_host
+          [-\w./]+    \s+                                 # target_host
+          [-\w./]+    \s+                                 # repository
+          [-\w./]+    \s+                                 # snapshot
+          \d+         \s+                                 # files
+          \d+         \s+                                 # files_recovered
+          \d+\.\d+%   \s+                                 # files_percent
+          \d+         \s+                                 # files_total
+          \d+         \s+                                 # bytes
+          \d+         \s+                                 # bytes_recovered
+          \d+\.\d+%   \s+                                 # bytes_percent
+          \d+         \s+                                 # bytes_total
+          -1          \s+                                 # translog_ops (always -1 for closed indices)
+          0           \s+                                 # translog_ops_recovered (always 0 for closed indices)
+          -1\.0%                                          # translog_ops_percent (always -1.0% for closed indices)
```

Comment:
Note: closed indices or frozen indices report `-1` values for translog stats because they don't recover operations from the translog. Maybe a value of `0` / `0.0%` would be more appropriate?

Reply:
Agreed, I think `0` operations is good. Also reporting `100.0%` would be good if the recovery is in stage `>= TRANSLOG`.

Reply:
Thanks @DaveCTurner. I opened https://github.com/elastic/elasticsearch/pull/39238 and I'll revisit this pull request when the upstream change will be merged.